picard/ui/mainwindow.py

```diff
@@ -1581,7 +1586,8 @@ def script_editor_closed(self):
         """Process "finished" signal from the script editor.
         """
         self.show_script_editor_action.setEnabled(True)
-        self.options_action.setEnabled(True)
+        self.script_editor_is_open = False
+        self.script_editor_dialog = None
```

Comment:
I really wonder if we need `script_editor_is_open` at all, what about just using `script_editor_dialog` ensuring it is `None` when dialog is closed (which should be the case already, right?).

Reply:
It's easy enough to get rid of it within the MainWindow dialog, but it is also used:
1. to indicate that the script editor has been opened in Options as in `picard/ui/options/renaming.py:L200` (in which case `self.script_editor_dialog` will still be `None`), and
2. to indicate that the script editor has been closed in Options as in `picard/ui/options/renaming.py:L203` (in which case `self.script_editor_dialog` will still be a `ScriptEditorDialog` instance).

In the second case, I suppose that we could have `picard/ui/options/renaming.py` force the MainWindow `self.script_editor_dialog` to `None` (with the appropriate cleanup actions), but I'm not sure how to handle the first case without potentially introducing more `ScriptEditorDialog` parent issues.  It may be something as simple as setting `self.script_editor_dialog` to `True` and open an instance if it is `not None`.  I'll have to think on that a bit more.

Reply:
I just tried it and it seems to work well.  I'll push an update.